src/library-aux/scala/Any.scala

```diff
@@ -110,13 +110,23 @@ abstract class Any {
    */
   final def ## : Int = sys.error("##")
 
-  /** Test whether the dynamic type of the receiver object is `T0`.
-   *
-   *  Note that the result of the test is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.isInstanceOf[String]` will return `false`, while the
-   *  expression `List(1).isInstanceOf[List[String]]` will return `true`.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the specified type.
+  /** Test whether the dynamic type of the receiver object has the same erasure to `T0`.
+   *
+   *  Depending on what `T0` is, the test is done in one of the below four ways:
+   *
+   *  * if T0 is a proper class type, e.g. `BigDecimal` this method returns `true` if
+   *    the value of the receiver object is a `BigDecimal` or a subtype of `BigDecimal`.
+   *  * if T0 is a parameterized class type, e.g. `List[Int]` this method returns `true` if
+   *    the value of the receiver object is some `List[X]` for any `X`.
+   *    For example, `List(1, 2, 3).isInstanceOf[List[String]]` will return true.
+   *  * if T0 is some singleton type `x.type` or `x` this method returns `this.eq(x)`.
+   *    For example, `x.isInstanceOf[1]` is equivalent to `x.eq(1)`
+   *  * if T0 is anything else, like a type parameter or an abstract type member, this method
+   *    returns true.
+   *    For example, `x.isInstanceOf[A]` where `A` is a type parameter will return true for
+   *    any value of `x`.
```

Comment:
Doesn't it emit an unchecked warning?  Worth mention imo.  It does for the type pattern.

Reply:
I decided against noting it's a warning because while I love elaborate documentation, not knowing that something will emit a warning is not a problem: the worst that can happen is that you are warned for something that you already know was the case. The likelier scenario is you come here after you see the warning.

No big deal though. If you prefer a comment that it warns, I'll add one.

Reply:
I added one